src/EndOfYearBanner/index.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useState } from 'react';
+import { ThemeProvider } from '@emotion/react';
+import {
+    Button,
+    buttonThemeReaderRevenueBrandAlt,
+    LinkButton,
+    SvgCross,
+    SvgInfo,
+} from '@guardian/source-react-components';
+import { OphanComponentEvent } from '@guardian/libs';
+
+import { BrazeClickHandler } from '../utils/tracking';
+import { styles as commonStyles } from '../styles/bannerCommon';
+import { styles } from './styles';
+import { canRender, COMPONENT_NAME } from './canRender';
+export { COMPONENT_NAME };
+
+export type BrazeMessageProps = {
+    header?: string;
+    body?: string;
+    boldText?: string;
+    buttonText?: string;
+};
+import type { ButtonTheme } from '@guardian/source-react-components';
+
+export type Props = {
+    logButtonClickWithBraze: BrazeClickHandler;
+    submitComponentEvent: (componentEvent: OphanComponentEvent) => void;
+    ophanComponentId?: string;
+    brazeMessageProps: BrazeMessageProps;
+};
+
+const catchAndLogErrors = (description: string, fn: () => void): void => {
+    try {
+        fn();
+    } catch (e) {
+        console.log(`Error (${description}): `, e.message);
+    }
+};
+
+const urlObject = new URL(
+    'https://www.theguardian.com/info/ng-interactive/2020/dec/21/the-guardian-in-2020',
+);
+
+const urlParams = {
+    INTCMP: 'gdnwb_mrtn_banner_edtrl_MK_SU_WorkingReport2020Canvas',
+};
+
+for (const [key, value] of Object.entries(urlParams)) {
+    urlObject.searchParams.set(key, value);
+}
+
+const THE_GU_IN_2020_URL = urlObject.href;
+
+const EndOfYearBanner: React.FC<Props> = (props: Props) => {
+    const {
+        logButtonClickWithBraze,
+        submitComponentEvent,
+        ophanComponentId = COMPONENT_NAME,
+        brazeMessageProps: { header, body, boldText, buttonText },
+    } = props;
+
+    const [showBanner, setShowBanner] = useState(true);
+
+    if (!canRender(props.brazeMessageProps)) {
+        return null;
+    }
+
+    if (!showBanner) {
+        return null;
+    }
+
+    const logToBrazeAndOphan = (internalButtonId: number): void => {
```

Comment:
Not necessarily as part of this PR, but we should think about abstracting this out. It's common across a few banners I think.